pkg/agent/controller/noderoute/node_route_controller.go

```diff
@@ -154,33 +146,20 @@ func (c *Controller) removeStaleGatewayRoutes() error {
 		return fmt.Errorf("error when listing Nodes: %v", err)
 	}
 
-	routes, err := c.routeClient.ListPeerCIDRRoute()
-	if err != nil {
-		return fmt.Errorf("error when listing existing gateway routes: %v", err)
-	}
-
+	// We iterate over all current Nodes, including the Node on which this agent is
+	// running, so the route to local Pods will be desired as well.
+	var desiredPodCIDRs []string
 	for _, node := range nodes {
-		_, podCIDR, err := net.ParseCIDR(node.Spec.PodCIDR)
-		if err != nil {
-			return fmt.Errorf("error when parsing Pod CIDR for Node %s: %v", node.Name, err)
-		}
-		podCIDRStr := podCIDR.String()
-		// remove from map of routes: at the end of the loop the remaining routes will be
-		// the stale ones (the ones we want to delete).
-		// we iterate over all current Nodes, including the Node on which this agent is
-		// running, so the route to local Pods will be removed from the map as well.
-		delete(routes, podCIDRStr)
-	}
-
-	// remove all remaining routes in the map: these routes do not match any Node in the
-	// cluster.
-	klog.V(4).Infof("Need to delete %d routes", len(routes))
-	for _, r := range routes {
-		if err := c.routeClient.DeletePeerCIDRRoute(r); err != nil {
-			klog.Errorf("error when deleting route '%v': %v", r, err)
+		if node.Spec.PodCIDR == "" {
```

Comment:
maybe add a comment for when this can happen

Reply:
Done